policies/module-types/template/tests/minijinja.rs

```diff
@@ -103,3 +108,188 @@ fn it_checks_correct_mini_jinja() {
     let output = read_to_string(&test_path).unwrap();
     assert_eq!("Hello World!", output);
 }
+
+#[test]
+fn test_minijinja_basename() {
+    assert_eq!(minijinja_filters::basename("".to_string()).unwrap(), "");
+    assert_eq!(minijinja_filters::basename("/".to_string()).unwrap(), "");
+    assert_eq!(minijinja_filters::basename("/.".to_string()).unwrap(), ".");
+    assert_eq!(
+        minijinja_filters::basename("/..".to_string()).unwrap(),
+        ".."
+    );
+    assert_eq!(minijinja_filters::basename(".".to_string()).unwrap(), ".");
+    assert_eq!(minijinja_filters::basename("..".to_string()).unwrap(), "..");
+    assert_eq!(
+        minijinja_filters::basename("afile..".to_string()).unwrap(),
+        "afile.."
+    );
+    assert_eq!(
+        minijinja_filters::basename("afile.".to_string()).unwrap(),
+        "afile."
+    );
+    assert_eq!(
+        minijinja_filters::basename("file.txt".to_string()).unwrap(),
+        "file.txt"
+    );
+    assert_eq!(
+        minijinja_filters::basename("/tmp/dir/file.".to_string()).unwrap(),
+        "file."
+    );
+    assert_eq!(
+        minijinja_filters::basename("/file.txt".to_string()).unwrap(),
+        "file.txt"
+    );
+    assert_eq!(
+        minijinja_filters::basename("/tmp/dir/file".to_string()).unwrap(),
+        "file"
+    );
+}
+
+#[test]
+fn test_basename_filter() {
+    let root_dir = tempdir().unwrap();
+    let test_path = root_dir.path().join("output");
+    fs::write(&test_path, "file").unwrap();
+
+    unix::test(
+        Path::new(BIN),
+        &format!(
+            r#"{{"path": "{}", "engine": "{}", "template_src": "{{{{ path|basename }}}}", "data": {{ "path": "/tmp/dir/file" }} }}"#,
+            test_path.display(),
+            "mini_jinja"
+        ),
+        PolicyMode::Audit,
+        Ok(Outcome::success()),
+    );
+    let output = read_to_string(&test_path).unwrap();
+    assert_eq!("file", output);
+}
+
+#[test]
+fn test_minijinja_dirname() {
+    assert_eq!(minijinja_filters::dirname("".to_string()).unwrap(), "");
+    assert_eq!(minijinja_filters::dirname("/".to_string()).unwrap(), "/");
+    assert_eq!(minijinja_filters::dirname(".".to_string()).unwrap(), "");
+    assert_eq!(minijinja_filters::dirname("..".to_string()).unwrap(), "");
+    assert_eq!(
+        minijinja_filters::dirname("/../..".to_string()).unwrap(),
+        "/.."
+    );
+    assert_eq!(
+        minijinja_filters::dirname("/file".to_string()).unwrap(),
+        "/"
+    );
+    assert_eq!(
+        minijinja_filters::dirname("/tmp/file".to_string()).unwrap(),
+        "/tmp"
+    );
+    assert_eq!(
+        minijinja_filters::dirname("/tmp/dir/file".to_string()).unwrap(),
+        "/tmp/dir"
+    );
+}
+
+#[test]
+fn test_dirname_filter() {
```

Comment:
the unit tests should be in the source file, this file is for integration tests (calling the agent)